build.sh

```diff
@@ -97,46 +97,32 @@ cd EmulatorJS
-compileProject "pcsx_rearmed" "https://github.com/EmulatorJS/pcsx_rearmed.git" "./" "Makefile.libretro"
-compileProject "picodrive" "https://github.com/EmulatorJS/picodrive.git" "./" "Makefile.libretro"
-compileProject "fbneo" "https://github.com/EmulatorJS/FBNeo.git" "./src/burner/libretro" "Makefile"
-compileProject "beetle-psx" "https://github.com/EmulatorJS/beetle-psx-libretro.git" "./" "Makefile"
-compileProject "beetle-pce" "https://github.com/EmulatorJS/beetle-pce-libretro.git" "./" "Makefile"
-compileProject "beetle-pcfx" "https://github.com/EmulatorJS/beetle-pcfx-libretro.git" "./" "Makefile"
-compileProject "beetle-ngp" "https://github.com/EmulatorJS/beetle-ngp-libretro.git" "./" "Makefile"
-compileProject "beetle-wswan" "https://github.com/EmulatorJS/beetle-wswan-libretro.git" "./" "Makefile"
-compileProject "gearcoleco" "https://github.com/EmulatorJS/Gearcoleco.git" "./platforms/libretro/" "Makefile"
-compileProject "parallel-n64" "https://github.com/EmulatorJS/parallel-n64.git" "./" "Makefile"
-compileProject "mame2003-plus" "https://github.com/EmulatorJS/mame2003-plus-libretro.git" "./" "Makefile"
-compileProject "puae" "https://github.com/EmulatorJS/libretro-uae.git" "./" "Makefile"
-compileProject "vice_x64" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=x64"
-compileProject "vice_x64sc" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=x64sc"
-compileProject "vice_x128" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=x128"
-compileProject "vice_xpet" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=xpet"
-compileProject "vice_xplus4" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=xplus4"
-compileProject "vice_xvic" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=xvic"
-compileProject "smsplus-gx" "https://github.com/EmulatorJS/smsplus-gx" "./" "Makefile.libretro"
+for row in $(jq -r '.[] | @base64' ../cores.json); do
+    _jq() {
+        echo ${row} | base64 --decode | jq -r ${1}
+    }
+
+    name=`echo $(_jq '.') | jq -r '.name'`
+    repo=`echo $(_jq '.') | jq -r '.repo'`
+    license=`echo $(_jq '.') | jq -r '.license'`
+    buildpath=`echo $(_jq '.') | jq -r '.makeoptions.buildpath'`
+    makescript=`echo $(_jq '.') | jq -r '.makeoptions.makescript'`
+    arguments=`echo $(_jq '.') | jq -r '.makeoptions.arguments[] | @base64'`
+
+    argumentstring=""
+    for rowarg in $(echo "${arguments}"); do
+        argumentstring="$argumentstring `echo $rowarg | base64 --decode`"
+    done
+
+    echo "Starting compile of core $name"
+
+    compileProject "$name" "$repo.git" "$buildpath" "$makescript" "$argumentstring"
+
+    if [ ! -z "$license" -a "$license" != " " ]; then
+        # license url is provided - download it
+        wget "$license?raw=true" -O "./$name-License.txt"
```

Comment:
I thought about that last night. I still want to leave the license URL in the cores.json file so we can link to it in place. But yes, this wget is sort of redundant. I'll work out a method to copy it rather than download it.